hoot-core/src/main/cpp/hoot/core/io/OsmApiWriter.cpp

```diff
@@ -165,18 +165,19 @@ bool OsmApiWriter::apply()
     //  Show the progress
     if (_showProgress)
     {
+      float percent_complete = _changeset.getProcessedCount() / (float)total;
       //  Actual progress is calculated and once it passes the next increment it is reported
-      if (_changeset.getProcessedCount() / (double)total * 100.0 >= progress + increment)
+      if (percent_complete >= progress + increment)
       {
-        progress += increment;
-        _progress.set((float)progress, "Uploading changeset...");
+        progress = percent_complete - fmod(percent_complete, increment);
+        _progress.set(percent_complete, "Uploading changeset...");
```

Comment:
While I was in here I fixed the progress for uploading changesets from 10000% back to the actual percentage.

Reply:
Nice!